shell: meta-keys handle condition update

jakub-uC opened the pull request:
Meta keys are active when they are enabled and when shell echo is set to on.
Updated meta keys description in shell's Kconfig file.

This change has been discussed with @aescolar in POSIX context.

Commit:
shell: meta-keys handle condition update

Meta keys are active when they are enabled and when shell echo is set
to on.
Updated meta keys description in shell's Kconfig file.

Signed-off-by: Jakub Rzeszutko <jakub.rzeszutko@nordicsemi.no>

nordic-krch reviewed (approved):
looks good with minor comment which is around this pr.

Commit:
shell: added static to char_replace function

Function char_replace is only used in shell_ops.c file.
Added keyword static to function definition and removed
declaration from shell_ops.h

Signed-off-by: Jakub Rzeszutko <jakub.rzeszutko@nordicsemi.no>

jarz-nordic commented on subsys/shell/shell_ops.c:
```diff
@@ -261,6 +261,11 @@ static void data_insert(const struct shell *shell, const char *data, u16_t len)
 void char_replace(const struct shell *shell, char data)
```
good catch - fixed
